core/src/main/java/apoc/periodic/Periodic.java

```diff
@@ -333,6 +341,9 @@ public Stream<BatchAndTotalResult> iterate(
                     retries,
                     result,
                     (tx, p) -> {
+                        if(tx instanceof InternalTransaction iTx){
+                            iTx.setMetaData(metaData);
+                        }
```

Comment:
Perhaps we should do this for all procedures that runs inner queries? Create a card for it at least?